src/java/org/streamingpool/core/service/InstrumentationService.java

```diff
@@ -0,0 +1,15 @@
+package org.streamingpool.core.service;
+
+import org.streamingpool.core.domain.StreamDependencyTree;
+
+/**
+ * Service to get instrumentation/debug information about a pool
+ */
+public interface InstrumentationService {
```

Comment:
Why have you decided to wrap the `StreamDependencyTree` in this `InstrumentationService`? Isn't it easier to just autowire everywhere the `StreamDependencyTree` directly and remove this service class?

Reply:
InstrumentationService is thought to be more general and provide wider interface int he future